mir/src/node.rs

```diff
@@ -227,16 +228,86 @@ impl MirNode {
                     .1
                     .expect("must have an absolute column ID on base"),
             },
-            MirNodeType::Reuse { ref node } => node.borrow().column_id_for_column(c),
+            MirNodeType::Reuse { ref node } => node.borrow().column_id_for_column(c, table_mapping.clone()),
             // otherwise, just look up in the column set
-            _ => match self.columns.iter().position(|cc| cc == c) {
+            _ =>
+            match self.columns.iter().position(|cc| cc == c) {
                 None => {
-                    panic!(
-                        "tried to look up non-existent column {:?} on node \"{}\" (columns: {:?})",
-                        c, self.name, self.columns
-                    );
-                }
-                Some(id) => id,
+                    // See if table mapping was passed in
+                    match table_mapping {
+                        // if mapping was passed in, then see if c has an associated table, and check
+                        // the mapping for a key based on this
+                        Some(map) => {
+                            match c.table.clone() {
```

Comment:
```suggestion
                            match *c.table {
```
... should avoid the `clone`.

Reply:
Done, but needed `Some(ref map)` instead